server/src/main/scala/drt/server/feeds/bhx/BHXArrivals.scala

```diff
@@ -0,0 +1,87 @@
+package drt.server.feeds.bhx
+
+import drt.shared.Arrival
+import javax.xml.datatype.XMLGregorianCalendar
+import org.joda.time.DateTime
+import services.SDate
+import uk.co.bhx.online.flightinformation.{FlightRecord, ScheduledFlightRecord}
+
+sealed trait BHXArrivals {
+
+  def convertToUTC(feedDate: XMLGregorianCalendar): Option[String] = {
+    val date = feedDate.toGregorianCalendar.getTime
+
+    date.getTime match {
+      case 0L => None
+      case _ =>
+        val datetime = new DateTime(date.getTime)
+
+        import org.joda.time.DateTimeZone
+        import org.joda.time.format.ISODateTimeFormat
+
+        Some(datetime.withZone(DateTimeZone.UTC).toString(ISODateTimeFormat.dateTime))
+    }
+  }
+
+}
+
+trait BHXLiveArrivals extends BHXArrivals {
+
+  def toLiveArrival(flightRecord: FlightRecord): Arrival =
+    new Arrival(Operator = "",
+      Status = flightRecord.getFlightStatus match {
+        case "Arrived" => "A"
+        case "Expected" => "E"
+        case "Cancelled" => "C"
+        case _ => "U"
+      },
+      EstDT = convertToUTC(flightRecord.getEstimatedTime).getOrElse(""),
+      ActDT = convertToUTC(flightRecord.getTouchdownTime).getOrElse(""),
+      EstChoxDT = convertToUTC(flightRecord.getEstimatedChoxTime).getOrElse(""),
+      ActChoxDT = convertToUTC(flightRecord.getChoxTime).getOrElse(""),
+      Gate = flightRecord.getGate,
+      Stand = flightRecord.getStand,
+      MaxPax = flightRecord.getCapacity,
+      ActPax = flightRecord.getPassengers,
+      TranPax = flightRecord.getTransits,
+      RunwayID = flightRecord.getRunway,
+      BaggageReclaimId = flightRecord.getBelt,
+      FlightID = 0,
+      AirportID = "BHX",
+      Terminal = flightRecord.getTerminal,
+      rawICAO = flightRecord.getFlightNumber,
```

Comment:
Is this correct?

Reply:
probably- we get two different types of flight numbers ICAO and IATA, but it's not very consistent which one we get from each feed. It's something we should probably put some effort into sorting out, but so far our solution is just to store whichever one we get in both fields.